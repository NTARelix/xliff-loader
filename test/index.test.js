const compiler = require('./compiler')

async function snapshotTestXliffFile(xliffPath) {
  const stats = await compiler(xliffPath)
  const output = stats.toJson().modules[0].source
  expect(output).toMatchSnapshot()
}

test('Transpiles simple xliff', () => snapshotTestXliffFile('./simple.xlf'))
test('Transpiles unicode xliff', () => snapshotTestXliffFile('./unicode.xlf'))
test('Transpiles special character xliff', () => snapshotTestXliffFile('./specialChars.xlf'))
test('Decodes HTML entities', () => snapshotTestXliffFile('./entities.xlf'))
