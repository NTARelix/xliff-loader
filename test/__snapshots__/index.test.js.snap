// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpiles simple xliff 1`] = `
"module.exports = {
  \\"res1\\": \\"Res 1\\",
  \\"res2\\": \\"Res 2\\",
  \\"res3\\": \\"Res 3\\"
}"
`;

exports[`Transpiles special character xliff 1`] = `
"module.exports = {
  \\"deAndre\\": \\"De'Andre\\",
  \\"html\\": \\"&lt;p&gt;Plan &amp; place content here&lt;p&gt;\\",
  \\"nonAsciiQuotes\\": \\"’“”\\"
}"
`;

exports[`Transpiles unicode xliff 1`] = `
"module.exports = {
  \\"hello\\": \\"今日は\\",
  \\"bulletinBoard\\": \\"掲示板\\",
  \\"butItDo\\": \\"人々はそれがそうであるとは思わないが、そうだ。\\",
  \\"doctorStory\\": \\"Le garçon de la forêt est devenu un médecin très réussi en Jamaïque.\\"
}"
`;
